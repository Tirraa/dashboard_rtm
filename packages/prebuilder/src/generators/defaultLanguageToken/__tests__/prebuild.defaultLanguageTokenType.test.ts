// eslint-disable-next-line import/no-extraneous-dependencies
import { afterAll, describe, expect, it } from 'vitest';

import generateDefaultLanguageTokenType from '../defaultLanguageTokenType';

// https://github.com/vitest-dev/vitest/discussions/2484
const fs = require('fs/promises');

const __TARGET_FOLDER_ROOT = './packages/prebuilder/src/generators/defaultLanguageToken/__tests__/FAKE_CODEGEN';
const __TARGET_FOLDER = __TARGET_FOLDER_ROOT + '/' + 'DEFAULT_LANGUAGE_TOKEN_TYPE';
const __TOKEN = 'FAKE_TOKEN';

describe('generateDefaultLanguageTokenType', () => {
  afterAll(async () => {
    await fs.rm(__TARGET_FOLDER, { recursive: true });
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_DEFAULT_LANGUAGE_TOKEN_TYPE';
    await generateDefaultLanguageTokenType(__TOKEN, targetFile, __TARGET_FOLDER);

    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});
