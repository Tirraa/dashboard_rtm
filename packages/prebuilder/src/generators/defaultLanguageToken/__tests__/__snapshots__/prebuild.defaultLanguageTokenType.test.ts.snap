// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateDefaultLanguageTokenType > should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
type FAKE_DEFAULT_LANGUAGE_TOKEN_TYPE = "FAKE_TOKEN";
export default FAKE_DEFAULT_LANGUAGE_TOKEN_TYPE;
"
`;

exports[`generateDefaultLanguageTokenType should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
type FAKE_DEFAULT_LANGUAGE_TOKEN_TYPE = "FAKE_TOKEN";
export default FAKE_DEFAULT_LANGUAGE_TOKEN_TYPE;
"
`;
